app/javascript/mastodon/components/status_quoted.tsx

```diff
@@ -0,0 +1,117 @@
+import { FormattedMessage } from 'react-intl';
+
+import classNames from 'classnames';
+
+import type { Map as ImmutableMap } from 'immutable';
+
+import { Icon } from 'mastodon/components/icon';
+import StatusContainer from 'mastodon/containers/status_container';
+import { useAppSelector } from 'mastodon/store';
+
+import QuoteIcon from '../../images/quote.svg?react';
+
+const QuoteWrapper: React.FC<{
+  isError?: boolean;
+  children: React.ReactNode;
+}> = ({ isError, children }) => {
+  return (
+    <div
+      className={classNames('status__quote', {
+        'status__quote--error': isError,
+      })}
+    >
+      <Icon id='quote' icon={QuoteIcon} className='status__quote-icon' />
+      {children}
+    </div>
+  );
+};
+
+type QuoteMap = ImmutableMap<'state' | 'quoted_status', string | null>;
```

Comment:
could we use an [Immutable Record](https://immutable-js.com/docs/v5.1.2/Record/) if this is a new data structure? or even better, not use Immutable at all?

Reply:
It comes through the status API/normalizer/reducer so difficult to refactor.